### PR TITLE
Web console: add explore view

### PR DESCRIPTION
**Pull request by vogievetsky**
This PR adds a simple, stateless, SQL backed, data exploration view to the web console. The idea is to let users explore data in Druid with point-and-click interaction and visualizations (instead of writing SQL and looking at a table). This can provide faster time-to-value for a user new to Druid and can allow a Druid veteran to quickly chart some data that they care about.

image

The view is self contained. There are only a handful of lines of code outside the `/explore-view/` path, and only to hook the view into the router and header.

The view is accessible via the header bar here:

image

### CODE REVIEW

**Review by clintropolis — APPROVED** [review]
this is rad :metal:

some general commentary on things i noticed (none are blockers since this is like totally new and marked as obviously experimental):
* man this is slick :rocket:
* filter bar allows clicking ok when no filtering is set, which makes bad queries (side note, it might be nice to show the bad query it tried to run to help us help you debug stuff, i had to dig around in my broker logs to figure out what was failing since i only saw the validation error message, but it wasn't obvious the query it was trying to run)
* filter bar maybe could use hover text or some branding to indicate that it is in fact for filtering stuff
* clicking 'show' on stuff that doesn't support grouping such as COMPLEX<json> silently has an error and doesn't update visual, had to dig around in web-console errors to find this
* same thing about trying to filter stuff that isn't supported for grouping
* picking a different table in the dropdown doesn't clear current state (filters, charts, etc) idk if it should... but things like filters it probably makes sense because it seems likely the same columns wouldn't exist 
* changing the 'stack by' does not seem to always update the chart.. though maybe this only happens when it gets into some other sort of error state 
* did i mention how rad this was? :tada:
* can has custom time granularity?
* maybe something funny going on when trying to filter time, since it doesn't seem to stick with the range im trying to choose
* definitely something funny going on when i try to pick an exact time interval, text box moves cursor to end and then cannot close if i can manage to type in fast enough to make the range i want
* since there is a default time filter, should it show up on the filter bar, at least if you are not in table view?

i'm so hyped i'll approve this even with these issues. will keep playing with stuff trying to find things

**Comment by clintropolis** [comment]
sweet looks like the things i ran into last time i tested have been fixed. I still can cause some silent errors that just have no indicator in the UI that something didn't work other than nothing changing, such as anything that would use a group query on a column type that isn't supported like json, but I haven't got it to run into any unrecoverable states yet.

:rocket:

File: docs/tutorials/tutorial-msq-convert-spec.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ description: How to convert an ingestion spec to a query for SQL-based ingestion
   ~ or more contributor license agreements.  See the NOTICE file
   ~ distributed with this work for additional information
   ~ regarding copyright ownership.  The ASF licenses this file
-  ~ to you under the Apache License, Version 2.0 (the
+  ~ to you under the Apache License, Version 2.0 (the
```
Comment (asdf2014) [inline docs/tutorials/tutorial-msq-convert-spec.md:13]:
```suggestion
  ~ to you under the Apache License, Version 2.0 (the
```

**Review by asdf2014 — APPROVED** [review]
🚀

**Pull request merged**